scripts/azureml-assets/azureml/assets/release/model_publish_utils/utils.py

```diff
@@ -0,0 +1,64 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+"""Model Utils Class."""
+
+from pathlib import Path
+from subprocess import PIPE, run, STDOUT
+import sys
+from azureml.assets.util import logger
+
+
+class ModelUtils:
+    """Download the Model from url at a given commit into the specified model directory."""
+
+    def __init__(self, model_url, model_commit_hash, model_dir):
+        """Create the base object for Model."""
+        self.model_url = model_url
+        self.model_commit_hash = model_commit_hash
+        self.model_dir = model_dir
+
+    def _run(self, cmd, cwd: Path = "./") -> int:
+        """Run the command and returns the result."""
+        logger.print(cmd)
+        result = run(
+            cmd,
+            cwd=cwd,
+            shell=True,
+            stdout=PIPE,
+            stderr=STDOUT,
+            encoding=sys.stdout.encoding,
+            errors="ignore",
+        )
+        if result.returncode != 0:
+            logger.print(f"Failed with error {result.stdout}.")
+        else:
+            logger.print(f"Successfully executed! Output: \n{result.stdout}")
+        return result.returncode
+
+    def _delete_downloaded_artifacts(self) -> None:
+        """Delete the downloaded artifacts in case of failure."""
+        logger.print("Deleting the incomplete model artifacts")
+        delete_cmd = f"rm -rf {self.model_dir}"
```

Comment:
> delete_cmd = f"rm -rf {self.model_dir}" [](http://example.com/codeflow?start=7&length=40)

`rm -rf` could have catastrophic effects.

Could we please create a temporary dir and mark it to be deleted instead?

Reply:
using temp directory.